modules/ROOT/pages/core-operations.adoc

```diff
@@ -300,51 +300,74 @@ Using the append and prepend methods may invalidate an existing JSON document.
 You can use xref:subdocument-operations.adoc[sub-document operations] if you want to have true JSON-aware prepend and append operations which add values to JSON arrays.
 ====
 
-[source,python]
+[source,java]
 ----
-append(docid, fragment)
-prepend(docid, fragment)
+append(document);
+prepend(document);
 ----
 
 The _append_ and _prepend_ operations atomically add bytes to the end or beginning of a binary document.
 They are an efficient alternative to retrieving a binary document in its entirety, appending the contents locally, and then saving the contents back to the server.
 
 Because these methods do raw string manipulation, they are only suitable for non-JSON documents: Prepending or appending anything to a JSON document will invalidate the JSON and make it unparseable by standard JSON parsers.
 
-The semantics of the _append_ and _prepend_ operations are similar to those of the _upsert_ family of operations, except that they accept the fragment to append as their value, rather than the entire document.
+The semantics of the _append_ and _prepend_ operations are similar to those of the _upsert_ family of operations, except that they accept the document whose content can be appended to the existing one, rather than the entire document.
 These functions may be used to add efficiency for custom binary data structures (such as logs), as they avoid transferring the contents of the entire document for each operation.
 Consider the following versions (which are equivalent)
 
 .Append using get() and replace() (slow)
-[source,python]
+[source,java]
 ----
-# Store the document
-cb.upsert('binary_doc', '\x01', format=couchbase.FMT_BYTES)
+// Create buffer out of a string
+ByteBuf toWrite = Unpooled.copiedBuffer("Hello", CharsetUtil.UTF_8);
+
+// Store the document
+bucket.upsert(BinaryDocument.create("binaryDoc", toWrite));
+
+// Get the document
+BinaryDocument read1 = bucket.get("binaryDoc", BinaryDocument.class);
+
+// Create a replacement document
+ByteBuf toAppend = Unpooled.copiedBuffer("World", CharsetUtil.UTF_8);
+byteBuf toReplace = read1.content().writeBytes(toAppend);
+
+// Replace the existing document
+bucket.replace("binaryDoc",BinaryDocument.create("binaryDoc", toReplace));
```

Comment:
I don't think this will compile.  Please do check that all examples compile and work.

Reply:
Apart from small `b` in `byteBuff` in line 332, all lines got compiled successfully.